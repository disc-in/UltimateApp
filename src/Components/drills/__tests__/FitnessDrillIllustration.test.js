import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { NavigationContext } from '@react-navigation/native';

import { createDrill } from '../../../Fixtures/TestFixtures';
import { DrillTypes } from '../../../Fixtures/config';

import FitnessDrillIllustration from '../FitnessDrillIllustration';

jest.mock('../../shared/VimeoVideo');

describe('<FitnessDrillIllustration />', () => {
  const startFitness = jest.fn();
  const steps = [
    {
      id: 1,
      repetition: 3,
      title: 'First Step',
      vimeoId: '406746924',
    },
    {
      id: 2,
      repetition: 3,
      title: 'Second Step',
      vimeoId: '406747302',
    },
    {
      id: 3,
      repetition: 3,
      title: 'Third Step',
      vimeoId: '406747476',
    },
  ];
  const drill = createDrill({ type: DrillTypes.FITNESS, title: 'Hot Box', steps });

  it('renders correctly with several steps', () => {
    const { toJSON } = render(<FitnessDrillIllustration drill={drill} startFitness={startFitness} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with one step', () => {
    const steps = [
      {
        id: 1,
        repetition: 3,
        title: 'First Step',
        vimeoId: '406746924',
      },
    ];
    const drill = createDrill({ type: DrillTypes.FITNESS, title: 'Hot Box', steps });

    // fake NavigationContext value data
    const navContext = {
      isFocused: () => true,
      // addListener returns an unscubscribe function.
      addListener: jest.fn(() => jest.fn()),
    };

    const { toJSON } = render(
      <NavigationContext.Provider value={navContext}>
        <FitnessDrillIllustration drill={drill} startFitness={startFitness} />
      </NavigationContext.Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('triggers startFitness', async () => {
    const { getByTestId } = render(<FitnessDrillIllustration drill={drill} startFitness={startFitness} />);

    fireEvent.press(getByTestId('startFitness'));

    expect(startFitness).toHaveBeenCalled();
  });
});
