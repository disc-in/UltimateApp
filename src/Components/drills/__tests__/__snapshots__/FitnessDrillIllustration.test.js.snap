// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FitnessDrillIllustration /> renders correctly with one step 1`] = `
<View
  style={
    Object {
      "height": 250,
      "marginBottom": 20,
    }
  }
>
  <View
    testID="VimeoVideoMock"
    vimeoId="406746924"
  />
</View>
`;

exports[`<FitnessDrillIllustration /> renders correctly with several steps 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingLeft": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#232b2b",
          "fontSize": 16,
        }
      }
    >
      • 
      3
       
      First Step
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#232b2b",
          "fontSize": 16,
        }
      }
    >
      • 
      3
       
      Second Step
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#232b2b",
          "fontSize": 16,
        }
      }
    >
      • 
      3
       
      Third Step
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginVertical": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2a9e91",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
          "width": "80%",
        }
      }
      testID="startFitness"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Start
      </Text>
    </View>
  </View>
</View>
`;
